tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc

```diff
@@ -5393,5 +5394,399 @@ TEST_F(EnergyPlusFixture, CoolingSimulate_WSHP_SourceSideOutletTemp)
+    // * Between the harcoded min/max defrost temperatures of 16F/-8.88C | 38F/3.33C
+    double constexpr oaTemp = 3.0;
+    state->dataEnvrn->OutDryBulbTemp = oaTemp;
+
+    double const oaWetbulb = Psychrometrics::PsyTwbFnTdbWPb(*state, oaTemp, 0.0, 101325.0);
+
+    Real64 constexpr expectedLoadMassFlowRate = 0.09999;
+    Real64 constexpr expectedCp = 4180;
+    Real64 constexpr specifiedLoadSetpoint = 45;
+
+    // This is not the case, even though the E+ I/O Documentation says it should
+    constexpr bool isLoadSideHeatTransferNegativeForCooling = false;
+
+    thisHeatingPLHP->onInitLoopEquip(*state, myHeatingLoadLocation);
+    {
+        EXPECT_ENUM_EQ(EIRFuelFiredHeatPump::OATempCurveVar::DryBulb, thisHeatingPLHP->oaTempCurveInputVar);
+        EXPECT_ENUM_EQ(EIRFuelFiredHeatPump::WaterTempCurveVar::EnteringCondenser, thisHeatingPLHP->waterTempCurveInputVar);
+
+        bool firstHVAC = true;
+        Real64 curLoad = 800;
+        bool runFlag = true;
+        Real64 const calculatedLoadInletTemp = specifiedLoadSetpoint - curLoad / (expectedLoadMassFlowRate * expectedCp);
+
+        state->dataLoopNodes->Node(thisHeatingPLHP->loadSideNodes.inlet).Temp = calculatedLoadInletTemp;
+        state->dataLoopNodes->Node(thisHeatingPLHP->loadSideNodes.outlet).TempSetPoint = specifiedLoadSetpoint;
+        state->dataLoopNodes->Node(thisHeatingPLHP->sourceSideNodes.inlet).Temp = oaTemp;
+
+        thisHeatingPLHP->simulate(*state, myHeatingLoadLocation, firstHVAC, curLoad, runFlag);
+        // expect it to meet setpoint and have some pre-evaluated conditions
+        // EXPECT_NEAR(specifiedLoadSetpoint, thisHeatingPLHP->loadSideOutletTemp, 0.001);
+        EXPECT_NEAR(curLoad, thisHeatingPLHP->loadSideHeatTransfer, 0.001);
+        {
+            ASSERT_GT(thisHeatingPLHP->capFuncTempCurveIndex, 0);
+            auto const *thisCurve = state->dataCurveManager->PerfCurve(thisHeatingPLHP->capFuncTempCurveIndex);
+            Real64 const waterTempforCurve = thisCurve->inputs[0];
+            Real64 const oaTempforCurve = thisCurve->inputs[1];
+            EXPECT_EQ(calculatedLoadInletTemp, waterTempforCurve);
+            EXPECT_EQ(oaTemp, oaTempforCurve);
+        }
+        {
+            ASSERT_GT(thisHeatingPLHP->powerRatioFuncTempCurveIndex, 0);
+            auto const *thisCurve = state->dataCurveManager->PerfCurve(thisHeatingPLHP->powerRatioFuncTempCurveIndex);
+            Real64 const waterTempforCurve = thisCurve->inputs[0];
+            Real64 const oaTempforCurve = thisCurve->inputs[1];
+            EXPECT_EQ(calculatedLoadInletTemp, waterTempforCurve);
+            EXPECT_EQ(oaTemp, oaTempforCurve);
+        }
+        {
+            ASSERT_GT(thisHeatingPLHP->defrostEIRCurveIndex, 0);
+            auto const *thisCurve = state->dataCurveManager->PerfCurve(thisHeatingPLHP->defrostEIRCurveIndex);
+            Real64 const oaTempforCurve = thisCurve->inputs[0];
+            EXPECT_EQ(oaTemp, oaTempforCurve);
+        }
+    }
```

Comment:
So I'm going to do 4 calls to simulate:

* For object in "Heating", "Cooling"
    * Simulate with Drybulb and Entering Condenser/Evaporator temperature
    * Simulate with Welbulb and Leaving Condenser/Evaporator temperature

For each of these, I check that the curves were evaluated with the proper conditions